source/app/libpd.c

```diff
@@ -344,6 +516,51 @@ static void get_parodus_url(char **parodus_url, char **client_url)
     	}
 }
 
+void parsePayloadForStatus(char *payload, char **cloudStatus)
+{
+	cJSON *json = NULL;
+	cJSON *cloudStatusObj = NULL;
+	char *cloud_status_str = NULL;
+
+	json = cJSON_Parse( payload );
+	if( !json )
+	{
+		WalError( "json parse error: [%s]\n", cJSON_GetErrorPtr() );
+	}
+	else
+	{
+		cloudStatusObj = cJSON_GetObjectItem( json, CLOUD_STATUS );
+		if( cloudStatusObj != NULL)
+		{
+			cloud_status_str = cJSON_GetObjectItem( json, CLOUD_STATUS )->valuestring;
+			if ((cloud_status_str != NULL) && strlen(cloud_status_str) > 0)
+			{
+				*cloudStatus = strdup(cloud_status_str);
+				WalPrint(" cloudStatus value parsed from payload is %s\n", *cloudStatus);
+			}
+			else
+			{
+				WalError("cloud status string is empty\n");
+			}
+		}
+		else
+		{
+			WalError("Failed to get cloudStatus from payload\n");
+		}
+		cJSON_Delete(json);
+	}
+}
+
+static void generate_trans_uuid(char **transID)
```

Comment:
Instead of double pointer argument, return the new string malloc pointer or NULL